### PR TITLE
Documents sub properly, adds subtract alias

### PR DESCRIPTION
**Pull request by mruberry**
`torch.sub` was undocumented, so this PR adds its documentation, analogous to `torch.add`'s documentation, and adds the alias `torch.subtract` for `torch.sub`, too. This alias comes from NumPy (see https://numpy.org/doc/stable/reference/generated/numpy.subtract.html?highlight=subtract#numpy.subtract)

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -6606,6 +6606,41 @@ def merge_dicts(*dicts):
...
+sub(input, other, *, alpha=1, out=None) -> Tensor
+
+Subtracts :attr:`other`, scaled by :attr:`alpha`, from :attr:`input`.
+
+.. math::
+    \text{{out}}_i = \text{{input}}_i - \text{{alpha}} \times \text{{other}}_i
+""" + r"""
+
+When :attr:`other` is a tensor, the shape of :attr:`other` must be
+:ref:`broadcastable <broadcasting-semantics>` with the shape of the underlying
```
Comment (ngimel) [inline torch/_torch_docs.py:6619]: what's underlying tensor?

**Review by ngimel — APPROVED** [review]

File: torch/_torch_docs.py (hunk 1 above)
Comment (mruberry) [reply torch/_torch_docs.py:6619]: Who knows?

Comment (mruberry) [reply torch/_torch_docs.py:6619]: Per offline conversation: this note will be modernized to "This op supports..." and include a broadcasting and type promotion reference.

**Comment by ezyang** [comment]
A request: please don't make unrelated whitespace changes in PRs; this just cost me another failed land (will have to try again tomorrow)

**Comment by mruberry** [comment]
Sorry about that @ezyang. I'll change my editor to not auto-truncate lines.

**Comment by ezyang** [comment]
Or better yet, add a lint rule to prevent trailing whitespace and banish it from the codebase entirely :)

**Pull request closed without merging**